elastic/apm-server
# add java attacher config

## Motivation/summary

Adding basic include/exclude and config flags.

Based on [this comment](https://github.com/elastic/apm-server/issues/4830#issuecomment-863207642) I pulled out version as something to be static for a given build of apm-server, which we update when we bundle with a new version of the attacher.

closes https://github.com/elastic/apm-server/issues/4829

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes



Update the `apm-server.yml` to include java attacher config fields, and verify they're parsed.

eg.
```yaml
  java_attacher:
    enabled: true
    continuous: true
    include_main:
      - "MyApplication"
      - "my-application.jar"
    exclude_user:
      - "root"
    include_vmargs:
      - "elastic.apm.agent.attach=true"
    config:
      - "service_name=my-cool-service"
      - "server_url=http://localhost:8200"
```

## Related issues
